orb/core/collection.py

```diff
@@ -344,7 +385,7 @@ def first(self, **context):
                     records = self.records(context=context)
                     record = records[0] if records else None
                 else:
-                    record = self._process([raw], context)[0]
+                    record = self._process([raw], context).next()
```

Comment:
using generators now to avoid unnecessarily creating large lists and then just getting a single index